doc/rst/developer/bestPractices/Valgrind.rst

```diff
@@ -0,0 +1,26 @@
+Valgrind
+========
+
+How-to 
+------
+
+To run valgrind on an executable generated by the chpl compiler:
+
+.. code-block:: bash
+
+     export CHPL_RT_NUM_THREADS_PER_LOCALE=100
+     export CHPL_TASKS=fifo
+     export CHPL_MEM=cstdlib
+     export CHPL_RE2_VALGRIND_SUPPORT=true
+
+     cd $CHPL_HOME
+     make
+     chpl <program.chpl>
+     valgrind ./<program>
+
+Explanation
+-----------
+
+Some third party libraries in the Chapel runtime will garble the output of
```

Comment:
I would try to describe this a little better than "garble". Something along the lines of:

"""
The above options are needed because not all third-party libraries support valgrind. In particular:
 - jemalloc doesn't doesn't support valgrind, which prevents valgrind from accurately tracking allocations/frees
 -  qthreads performs task-switching in user-space, so valgrind isn't able to keep track of the stack frame correctly
 - re2 intentionally leaves some memory uninitialized for performance reasons (unless `CHPL_RE2_VALGRIND_SUPPORT=true` is set at build time)
 - GASNet support for valgrind is experimental at this time -- see https://github.com/chapel-lang/chapel/issues/8544 for the current status

"""